### PR TITLE
Expose configuredSharedTree

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Expose an `@internal` api, `configuredSharedTree` which can be used to opt into various internal/debug shared tree settings.

This should help cases like https://github.com/microsoft/FluidFramework/issues/20274 root cause the bug.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: examples/data-objects/inventory-app/src/reactSharedTreeView.tsx [hunk 1]
```diff
@@ -10,18 +10,29 @@ import {
...
+} from "@fluidframework/tree/internal";
 import * as React from "react";
 
 /**
  * This file contains logic not specific to this particular sample that other apps may want to use.
  * Eventually this should be published as part of a package apps can use.
  */
 
+/**
+ * Opt into extra validation to detect detect encoding bugs and data corruption.
```
Comment (yann-achard-MS) [inline examples/data-objects/inventory-app/src/reactSharedTreeView.tsx:30]:
```suggestion
 * Opt into extra validation to detect encoding bugs and data corruption.
```

File: packages/dds/tree/src/treeFactory.ts [hunk 2]
```diff
@@ -51,17 +51,44 @@ export class TreeFactory implements IChannelFactory<ITree> {
...
+ * @remarks
+ * This is useful for debugging and testing to opt into extra validation or see if opting out of some optimizations fixes an issue.
+ * @example
+ * ```typescript
+ * import {
+ * 	ForestType,
+ * 	TreeCompressionStrategy,
+ * 	configuredSharedTree,
+ * 	typeboxValidator,
+ * 	// eslint-disable-next-line import/no-internal-modules
```
Comment (yann-achard-MS) [inline packages/dds/tree/src/treeFactory.ts:69]: Does this eslint directive need to be in the example?

Comment (CraigMacomber) [reply packages/dds/tree/src/treeFactory.ts:69]: I considered removing it. I kinda like that it helps highlight that despite this pattern usually not being allowed, in this case (reaching into a package for on of our release tag rollups) is supported.

I don't have a strong opinion either way though, so if anyone has a preference, I'll go with that.

Comment (yann-achard-MS) [reply packages/dds/tree/src/treeFactory.ts:69]: No strong opinion either. Feel free to resolve without further changes.

**Review by yann-achard-MS — APPROVED** [review]

**Pull request merged**